lib/improver/wxcode/weather_symbols.py

```diff
@@ -260,18 +278,28 @@ def construct_extract_constraint(diagnostics, thresholds):
             iris.Constraint or list of iris.Constraints:
                 The constructed iris constraints.
         """
+        def _constraint_string(diagnostic, threshold):
```

Comment:
I'm not sure about functions within functions. I would prefer constraint_str = ....  and return constraint_str but I'm not sure this is actually a problem.

Reply:
I'll leave this if there is no strong objection. I could just use a format statement I suppose, but I wanted a doc string to make clear what is going on.